src/main/java/com/google/cloud/spanner/pgadapter/utils/Converter.java

```diff
@@ -27,11 +29,62 @@
 import com.google.cloud.spanner.pgadapter.parsers.NumericParser;
 import com.google.cloud.spanner.pgadapter.parsers.StringParser;
 import com.google.cloud.spanner.pgadapter.parsers.TimestampParser;
+import com.google.cloud.spanner.pgadapter.statements.IntermediateStatement;
 import com.google.common.base.Preconditions;
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
 
 /** Utility class for converting between generic PostgreSQL conversions. */
 public class Converter {
-  private Converter() {}
+  private final ByteArrayOutputStream buffer = new ByteArrayOutputStream(256);
+  private final DataOutputStream outputStream = new DataOutputStream(buffer);
+  private final IntermediateStatement statement;
+  private final QueryMode mode;
+  private final OptionsMetadata options;
+  private final ResultSet resultSet;
+
+  public Converter(
+      IntermediateStatement statement,
+      QueryMode mode,
+      OptionsMetadata options,
+      ResultSet resultSet) {
+    this.statement = statement;
+    this.mode = mode;
+    this.options = options;
+    this.resultSet = resultSet;
+  }
+
+  public int getColumnCount() {
+    return resultSet.getColumnCount();
+  }
+
+  public ResultSet getResultSet() {
+    return resultSet;
+  }
+
+  public int convertResultSetRowToDataRowResponse() throws IOException {
+    buffer.reset();
+    outputStream.writeShort(resultSet.getColumnCount());
+    for (int column_index = 0; /* column indices start at 0 */
+        column_index < resultSet.getColumnCount();
```

Comment:
Can we use the `getColumnCount` method created above here or remove that method as its not used anywhere?

Reply:
Good point. Removed the redundant `getColumnCount()` method.